libvast/src/system/make_source.cpp

```diff
@@ -0,0 +1,148 @@
+  // Parse schema local to the import command.
+  auto schema = get_schema(options);
+  if (!schema)
+    return schema.error();
+  // Discern the input source (file, stream, or socket).
+  if (uri && file)
+    return caf::make_error(ec::invalid_configuration, //
+                           "only one source possible (-r or -l)");
+  if (!uri && !file)
+    file = std::string{defaults::import::read};
+  if (uri) {
+    endpoint ep;
+    if (!vast::parsers::endpoint(*uri, ep))
+      return caf::make_error(vast::ec::parse_error, "unable to parse endpoint",
+                             *uri);
+    if (!ep.port)
+      return caf::make_error(vast::ec::invalid_configuration,
+                             "endpoint does not "
+                             "specify port");
+    if (ep.port->type() == port_type::unknown)
+      // Fall back to tcp if we don't know anything else.
+      ep.port = port{ep.port->number(), port_type::tcp};
+    VAST_INFO("{}-reader listens for data on {}", format,
+              ep.host + ":" + to_string(*ep.port));
+    switch (ep.port->type()) {
+      default:
+        return caf::make_error(vast::ec::unimplemented,
+                               "port type not supported:", ep.port->type());
+      case port_type::udp:
+        udp_port = ep.port->number();
+        break;
+    }
+  }
+  auto reader = format::reader::make(format, inv.options);
+  if (!reader)
+    return reader.error();
+  if (slice_size == std::numeric_limits<decltype(slice_size)>::max())
+    VAST_VERBOSE("{} produces {} table slices", (*reader)->name(), encoding);
+  else
+    VAST_VERBOSE("{} produces {} table slices of at most {} events",
+                 (*reader)->name(), encoding, slice_size);
+  // Spawn the source, falling back to the default spawn function.
+  auto local_schema = schema ? std::move(*schema) : vast::schema{};
+  auto type_filter = type ? std::move(*type) : std::string{};
+  auto src =
+    [&](auto&&... args) {
+      if (udp_port) {
+        if (detached)
+          return sys.middleman().spawn_broker<caf::spawn_options::detach_flag>(
+            datagram_source, *udp_port, std::forward<decltype(args)>(args)...);
+        return sys.middleman().spawn_broker(
+          datagram_source, *udp_port, std::forward<decltype(args)>(args)...);
+      }
+      if (detached)
+        return sys.spawn<caf::spawn_options::detach_flag>(
+          source, std::forward<decltype(args)>(args)...);
+      return sys.spawn(source, std::forward<decltype(args)>(args)...);
+    }(std::move(*reader), slice_size, max_events, std::move(type_registry),
+      std::move(local_schema), std::move(type_filter), std::move(accountant));
+  VAST_ASSERT(src);
+  // Attempt to parse the remainder as an expression.
+  if (!inv.arguments.empty()) {
+    auto expr = parse_expression(inv.arguments.begin(), inv.arguments.end());
+    if (!expr)
+      return expr.error();
+    anon_send(src, std::move(*expr));
+  }
```

Comment:
Side note: Didn't we deprecate this?

Reply:
Still exists the same way in the current `make_source.hpp`.

Reply:
I just think that `inv.arguments.size() <= 1` is always true. Anyways, we can always change this some other time.